pkg/services/serviceaccounts/api/token_test.go

```diff
@@ -238,3 +238,101 @@ func TestServiceAccountsAPI_DeleteToken(t *testing.T) {
+	sa := tests.SetupUserServiceAccount(t, store, tests.TestUser{Login: "sa", IsServiceAccount: true})
+
+	type testCreateSAToken struct {
+		desc                      string
+		keyName                   string
+		secondsToLive             int64
+		expectedHasExpired        bool
+		expectedResponseBodyField string
+		expectedCode              int
+		acmock                    *accesscontrolmock.Mock
+	}
+
+	testCases := []testCreateSAToken{
+		{
+			desc:          "should be able to list serviceaccount with no expiration date",
+			keyName:       "Test1",
+			secondsToLive: 0,
+			acmock: tests.SetupMockAccesscontrol(
+				t,
+				func(c context.Context, siu *models.SignedInUser, _ accesscontrol.Options) ([]*accesscontrol.Permission, error) {
+					return []*accesscontrol.Permission{{Action: serviceaccounts.ActionRead, Scope: "serviceaccounts:id:1"}}, nil
+				},
+				false,
+			),
+			expectedHasExpired:        false,
+			expectedResponseBodyField: "hasExpired",
+			expectedCode:              http.StatusOK,
+		},
+		{
+			desc:          "should be able to list serviceaccount with secondsUntilExpiration",
+			keyName:       "Test2",
+			secondsToLive: 1000,
+			acmock: tests.SetupMockAccesscontrol(
+				t,
+				func(c context.Context, siu *models.SignedInUser, _ accesscontrol.Options) ([]*accesscontrol.Permission, error) {
+					return []*accesscontrol.Permission{{Action: serviceaccounts.ActionRead, Scope: "serviceaccounts:id:1"}}, nil
+				},
+				false,
+			),
+			expectedHasExpired:        false,
+			expectedResponseBodyField: "secondsUntilExpiration",
+			expectedCode:              http.StatusOK,
+		},
+		{
+			desc:          "should be able to list serviceaccount with expired token",
+			keyName:       "Test3",
+			secondsToLive: 1,
+			acmock: tests.SetupMockAccesscontrol(
+				t,
+				func(c context.Context, siu *models.SignedInUser, _ accesscontrol.Options) ([]*accesscontrol.Permission, error) {
+					return []*accesscontrol.Permission{{Action: serviceaccounts.ActionRead, Scope: "serviceaccounts:id:1"}}, nil
+				},
+				false,
+			),
+			expectedHasExpired:        true,
+			expectedResponseBodyField: "secondsUntilExpiration",
+			expectedCode:              http.StatusOK,
+		},
+	}
+
+	var requestResponse = func(server *web.Mux, httpMethod, requestpath string, requestBody io.Reader) *httptest.ResponseRecorder {
+		req, err := http.NewRequest(httpMethod, requestpath, requestBody)
+		require.NoError(t, err)
+		req.Header.Add("Content-Type", "application/json")
+		recorder := httptest.NewRecorder()
+		server.ServeHTTP(recorder, req)
+		return recorder
+	}
+
+	for i, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			_ = createTokenforSA(t, tc.keyName, sa.OrgId, sa.Id, tc.secondsToLive)
+
+			endpoint := fmt.Sprintf(serviceaccountIDPath+"/tokens", sa.Id)
+			bodyString := ""
+			server := setupTestServer(t, &svcmock, routing.NewRouteRegister(), tc.acmock, store)
+			if i == 2 {
+				time.Sleep(time.Millisecond * 1000)
+			}
```

Comment:
this is something I am not proud of.

---
I have to wait 1 second for making a token expired.
This is going to delay all builds by 1 sec